pkg/virt-config/configuration_test.go

```diff
@@ -0,0 +1,630 @@
+		Entry("when unset, GetSELinuxLauncherType should return the default", virtconfig.DefaultSELinuxLauncherType, virtconfig.DefaultSELinuxLauncherType),
+	)
+
+	DescribeTable(" when OVMFPath", func(cpuArch string, ovmfPathKey string, result string) {
+
+		kv := &v1.KubeVirt{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kubevirt",
+				Namespace: "kubevirt",
+			},
+			Spec: v1.KubeVirtSpec{
+				Configuration: v1.KubeVirtConfiguration{
+					OVMFPath: ovmfPathKey,
+				},
+			},
+			Status: v1.KubeVirtStatus{
+				Phase: "Deployed",
+			},
+		}
+
+		clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVWithCPUArch(kv, cpuArch)
+		ovmfPath := clusterConfig.GetOVMFPath()
+		Expect(ovmfPath).To(Equal(result))
+	},
+		Entry("when set, GetOVMFPath should return the value", "", "/usr/share/ovmf/x64", "/usr/share/ovmf/x64"),
+		Entry("when unset, GetOVMFPath should return the default with amd64", "amd64", "", virtconfig.DefaultARCHOVMFPath),
+		Entry("when unset, GetOVMFPath should return the default with arm64", "arm64", "", virtconfig.DefaultAARCH64OVMFPath),
+		Entry("when unset, GetOVMFPath should return the default with ppc64le", "ppc64le", "", virtconfig.DefaultARCHOVMFPath),
+	)
+
+	It("verifies that SetConfigModifiedCallback works as expected ", func() {
+		lock := &sync.Mutex{}
+		var callbackSet1, callbackSet2 bool
+		callback1 := func() {
+			lock.Lock()
+			defer lock.Unlock()
+			callbackSet1 = true
+		}
+		callback2 := func() {
+			lock.Lock()
+			defer lock.Unlock()
+			callbackSet2 = true
+		}
+		KV := &v1.KubeVirt{
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: rand.String(10),
+				Name:            "kubevirt",
+				Namespace:       "kubevirt",
+			},
+			Spec: v1.KubeVirtSpec{
+				Configuration: v1.KubeVirtConfiguration{
+					DeveloperConfiguration: &v1.DeveloperConfiguration{
+						LogVerbosity: &v1.LogVerbosity{
+							VirtLauncher: 3,
+						},
+					},
+				},
+			},
+			Status: v1.KubeVirtStatus{
+				Phase: v1.KubeVirtPhaseDeploying,
+			},
+		}
+		clusterConfig, _, kubeVirtInformer := testutils.NewFakeClusterConfigUsingKV(KV)
+		callbackSet1 = false
+		callbackSet2 = false
+		clusterConfig.SetConfigModifiedCallback(callback1)
+		clusterConfig.SetConfigModifiedCallback(callback2)
+
+		Expect(clusterConfig.GetVirtLauncherVerbosity()).To(Equal(uint(3)))
+		KV.Spec.Configuration.DeveloperConfiguration.LogVerbosity.VirtLauncher = 6
+		testutils.UpdateFakeKubeVirtClusterConfig(kubeVirtInformer, KV)
+		Expect(clusterConfig.GetVirtLauncherVerbosity()).To(Equal(uint(6)))
+		Eventually(func() bool {
+			lock.Lock()
+			defer lock.Unlock()
+			return callbackSet1 && callbackSet2
+		}).Should(BeTrue())
+	})
+
+	It("Should still get GetPermittedHostDevices after invalid update", func() {
+		//expectedDevices := `{"pciHostDevices":[{"pciVendorSelector":"10DE:1EB8","resourceName":"nvidia.com/TU104GL_Tesla_T4"}],"mediatedDevices":[{"mdevNameSelector":"GRID T4-1Q","resourceName":"nvidia.com/GRID_T4-1Q"}]}`
```

Comment:
also here the test tries to verify that invalid configs return the last valid values.